### PR TITLE
#11 Adds posts for selection sort

### PR DESCRIPTION
**Pull request by niyoj**
# Fixes issue number #11 

## Summary of the Changes
- The issue here was to add new blog posts for selection sort under the `dsa` category,
- First, the new author file `niyoj.md` was created inside `_authors` directory,
- Then, the avatar was added for the author inside `assets/images/author` named as `niyoj.jpg`,
- Since no posts were written previously, there was no `_posts` folder so, `_posts` folder was created,
- A new file `2021-10-01-selection-sort.md` was added inside the folder `_posts` which was the blog posts written.

### CODE REVIEW

File: _posts/2021-10-01-selection-sort.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
+---
+title: Selection Sort
+category: dsa
+layout: post
+author: niyoj
+avatar: "/assets/images/author/niyoj.jpg"
+---
+
+Sorting algorithms are the algorithms that are used to sort any given data in a dataset in ascending or descending order. There are various search algorithms like the bubble sort, quick sort, etc. among which one of the simpler and common types of the sort is selection sort.
```
Comment (Ruchip16) [inline _posts/2021-10-01-selection-sort.md:9]: Here there are various *sort* algorithms would come instead of *search*

**Review by Ruchip16 — CHANGES_REQUESTED** [review]

**Comment by Ruchip16** [comment]
@niyoj make the necessary changes and then you can just rebase your commit in this PR itself, feel free to ask doubts if any

**Comment by niyoj** [comment]
I corrected the mistakes.can you look it once.

**Review by Ruchip16 — APPROVED** [review]

**Comment by Ruchip16** [comment]
@LoopThrough-i-j can you please review this once before merging?

**Comment by LoopThrough-i-j** [comment]
Please read the docs check [this](https://github.com/Mentro-Org/Mentro-Community-Blog/blob/main/contributing.md#editting) for how to add authors and posts. Needs changes in the config file, and your post metadata.

**Comment by Ruchip16** [comment]
> Please read the docs check [this](https://github.com/Mentro-Org/Mentro-Community-Blog/blob/main/contributing.md#editting) for how to add authors and posts. Needs changes in the config file, and your post metadata.

@niyoj plz make the required changes suggested by @LoopThrough-i-j

**Comment by niyoj** [comment]
Ok, WIll do that.

**Comment by niyoj** [comment]
I made all the changes as suggested.

**Comment by niyoj** [comment]
@Ruchip16 Can you merge this with main branch now?

**Comment by niyoj** [comment]
@LoopThrough-i-j Can you please review my PR once more? I have updated as you have mentioned.

File: _config.yml [hunk 2]
```diff
@@ -36,6 +36,13 @@ authors:
     bio: "Co-founder Mentro"
     email: "debayan@mentro.tech"
     linkedin: https://www.linkedin.com/in/debayan-ganguly/
+  niyoj:
+    name: "Niyoj Oli"
+    site: "https://niyojoli.com.np"
+    avatar: "/assets/images/author/niyoj.jpg"
+    bio: "Open source contributor"
+    email: "oliniyoj@gmail.com"
+    linkedin: https://www.linkedin.com/in/niyoj
```
Comment (LoopThrough-i-j) [inline _config.yml:46]: you might also add your `twitter` and `medium` links if you have some.

File: _posts/2021-10-01-selection-sort.md [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+title: "Selection Sort - A simple sorting algorithm"
+description: "This blog post is about Selection Sort, a simple sorting algorithm. Here, we will talk more about selection sort, how it works, and finally look at its pseudo-code for its implementation."
+keywords: 
+    - Mentro
+    - Selection Sort
+    - Sorting Algorithm
+category: dsa
+layout: post
+author: niyoj
+avatar: "/assets/images/author/niyoj.jpg"
```
Comment (LoopThrough-i-j) [inline _posts/2021-10-01-selection-sort.md:11]: Not sure if the avatar is required.

File: _posts/2021-10-01-selection-sort.md [hunk 4]
```diff
@@ -0,0 +1,155 @@
+---
+title: "Selection Sort - A simple sorting algorithm"
+description: "This blog post is about Selection Sort, a simple sorting algorithm. Here, we will talk more about selection sort, how it works, and finally look at its pseudo-code for its implementation."
+keywords: 
+    - Mentro
+    - Selection Sort
+    - Sorting Algorithm
+category: dsa
+layout: post
```
Comment (LoopThrough-i-j) [inline _posts/2021-10-01-selection-sort.md:9]: The layout is not required.

File: _posts/2021-10-01-selection-sort.md [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+author: niyoj
+avatar: "/assets/images/author/niyoj.jpg"
+tags:
+    - sorting algorithm
+    - dsa
+    - selection sort
+permalink: /dsa/:title/
+image: /assets/images/b1/mentor.jpg
```
Comment (LoopThrough-i-j) [inline _posts/2021-10-01-selection-sort.md:17]: For this either use the default image for DSA or upload a new image and link it.

**Comment by LoopThrough-i-j** [comment]
@niyoj Thanks for the post added a few more comments.

File: _config.yml (hunk 2 above)
Comment (niyoj) [reply _config.yml:46]: I don't use twitter so I didn't mentioned them here.

File: _posts/2021-10-01-selection-sort.md (hunk 3 above)
Comment (niyoj) [reply _posts/2021-10-01-selection-sort.md:11]: Ok will remove that too.

File: _posts/2021-10-01-selection-sort.md (hunk 4 above)
Comment (niyoj) [reply _posts/2021-10-01-selection-sort.md:9]: Ok. Will remove that and send PR once again.

File: _posts/2021-10-01-selection-sort.md (hunk 5 above)
Comment (niyoj) [reply _posts/2021-10-01-selection-sort.md:17]: Ok will consider that too.

**Comment by niyoj** [comment]
Are there any additional changes to be made?

**Comment by niyoj** [comment]
I mistakenly deleted the forked repo. So, I have to open another PR.

**Pull request closed without merging**